alkira/resource_alkira_checkpoint.go

```diff
@@ -0,0 +1,326 @@
+							Required:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+						"reachability": {
+							Description: "This option specifies whether the management server is " +
+								"publicly reachable or not. If the reachability is private then you " +
+								"need to provide the segment to be used to access the management server. ",
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"segment_id": {
+							Description: "The id of the segment to be used to access the management server.",
+							Type:        schema.TypeInt,
+							Optional:    true,
+						},
+						"type": {
+							Description: "The type of the management server.",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"user_name": {
+							Description: "The user_name of the management server.",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+					},
+				},
+			},
+			"max_instance_count": {
+				Description: "The maximum number of Checkpoint Firewall instances that should be " +
+					"deployed when auto-scale is enabled. Note that auto-scale is not supported " +
+					"with Checkpoint at this time. max_instance_count must be greater than or " +
+					"equal to min_instance_count.",
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"min_instance_count": {
+				Description: "The minimum number of Checkpoint Firewall instances that should be " +
+					"deployed at any point in time.",
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  0,
+			},
+			"name": {
+				Description: "Name of the Checkpoint Firewall service.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"pdp_ips": {
+				Description: "The IPs of the PDP Brokers.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"segment_names": {
+				Description: "The names of the segments associated with the service.",
+				Type:        schema.TypeList,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"segment_options": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"segment_id": {
+							Description: "The ID of the segment.",
+							Type:        schema.TypeInt,
+							Required:    true,
+						},
+						"zone_name": {
+							Description: "The name of the associated zone.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"groups": {
+							Description: "The list of Groups associated with the zone.",
+							Type:        schema.TypeList,
+							Required:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+			"size": {
+				Description:  "The size of the service, one of `SMALL`, `MEDIUM`, `LARGE`, `2LARGE`, `4LARGE`.",
```

Comment:
Could you confirm the size from the portal? I thought we never allowed 4LARGE yet... even 2LARGE... don't think we support it...